generator-theia/src/extension/extension-package-generator.ts

```diff
@@ -101,4 +104,26 @@ export class ExtensionPackageGenerator extends AbstractGenerator {
         }
     }
 
-}
\ No newline at end of file
+    protected compilePackageTdOptions(): object {
+        return {
+            "excludeNotExported": true,
+            "exclude": "lib, node_modules",
+            "experimentalDecorators": true,
+            "ignoreCompilerErrors": true,
+            "mode": "file",
+            "out": "./docs/" + this.getPackageName(),
```

Comment:
You can use this.model.pck to get the name 
see:
`this.model.pck = this.fs.readJSON(`extension.package.json`) || {};`
In extension-generator.ts